lib/plug/session/cookie.ex

```diff
@@ -218,4 +198,55 @@ defmodule Plug.Session.COOKIE do
 
   defp check_serializer(_),
     do: raise(ArgumentError, "cookie store expects :serializer option to be a module")
+
+  defp read_raw_cookie(raw_cookie, opts) do
+    signing_salt = derive(opts.secret_key_base, opts.signing_salt, opts.key_opts)
+
+    case opts do
+      %{encryption_salt: nil} ->
+        MessageVerifier.verify(raw_cookie, signing_salt)
+
+      %{encryption_salt: _} ->
+        encryption_salt = derive(opts.secret_key_base, opts.encryption_salt, opts.key_opts)
+
+        MessageEncryptor.decrypt(raw_cookie, encryption_salt, signing_salt)
+    end
+    |> case do
+      :error -> nil
+      result -> result
+    end
+  end
+
+  defp build_opts(opts) do
+    encryption_salt = opts[:encryption_salt]
+    signing_salt = check_signing_salt(opts)
+
+    iterations = Keyword.get(opts, :key_iterations, 1000)
+    length = Keyword.get(opts, :key_length, 32)
+    digest = Keyword.get(opts, :key_digest, :sha256)
+    log = Keyword.get(opts, :log, :debug)
+    secret_key_base = Keyword.get(opts, :secret_key_base)
+    key_opts = [iterations: iterations, length: length, digest: digest, cache: Plug.Keys]
+
+    serializer = check_serializer(opts[:serializer] || :external_term_format)
+
+    %{
+      secret_key_base: secret_key_base,
+      encryption_salt: prederive(secret_key_base, encryption_salt, key_opts),
+      signing_salt: prederive(secret_key_base, signing_salt, key_opts),
+      key_opts: key_opts,
+      serializer: serializer,
+      log: log
+    }
+  end
+
+  defp build_rotating_opts(opts, rotating_opts) when is_list(rotating_opts) do
+    Map.put(opts, :rotating_options, Enum.map(rotating_opts, &build_opts/1))
+  end
+
+  defp build_rotating_opts(opts, {module, function, args}) do
```

Comment:
Do we need the MFA support? If so, generally speaking the MFA is invoked at runtime, not during `init`. But if we don't have a use case, better to remove it for now. :)

Reply:
Makes sense and I don't have any use case. I will remove it from here and the mention in the docs :+1: